asset/gulpfile.js

```diff
@@ -40,39 +37,65 @@ var gulp = require('gulp'),
         'yaml'
     ];
 
+// Images
 gulp.task('images', function () {
     return gulp.src('img/*')
         .pipe(gulp.dest('../theme/img/'));
 });
 
+// Icons
+gulp.task('icons', function () {
+    return gulp.src('node_modules/font-awesome/fonts/**.*')
+        .pipe(gulp.dest('../theme/css/fonts'));
+});
+
+// JavaScript
 gulp.task('scripts', function () {
     var prismComponents = [];
     for (var component in prism) {
-        prismComponents[component] = 'node_modules/prismjs/components/prism-' + prism[component] + '.min.js';
+        prismComponents[component] = 'node_modules/prismjs/components/prism-' + prism[component] + '.js';
     }
 
-    return gulp.src(prismComponents.concat([
-            'node_modules/prismjs/plugins/normalize-whitespace/prism-normalize-whitespace.min.js',
-            'node_modules/anchor-js/anchor.min.js',
-            'js/base.js',
-            'js/scripts.js'
-        ]))
-        .pipe(concat({path: 'scripts.js'}))
-        .pipe(uglify({mangle: false}))
+    return gulp.src(
+        prismComponents.concat(
+            [
+                'node_modules/prismjs/plugins/normalize-whitespace/prism-normalize-whitespace.min.js',
+                'node_modules/jquery/dist/jquery.min.js',
+                'node_modules/bootstrap-sass/assets/javascripts/bootstrap.min.js',
+                'node_modules/anchor-js/anchor.min.js',
+                'js/base.js',
+                'js/component-list.js'
+            ]
+        )
+    )
+        .pipe(concat({path : 'scripts.js'}))
+        .pipe(uglify({mangle : false}))
         .pipe(rev())
         .pipe(gulp.dest('../theme/js/'))
-        .pipe(rev.manifest('../assets.yml', {merge: true}))
+        .pipe(rev.manifest('../assets.yml', {merge : true}))
         .pipe(gulp.dest('./'));
 });
 
+// CSS
 gulp.task('styles', function () {
-    return gulp.src('sass/styles.scss')
-        .pipe(cssimport({filter: /^..\/node_modules\//gi}))
-        .pipe(sass({outputStyle: 'compressed'}))
+    return gulp.src(
+        [
+            'node_modules/prismjs/themes/prism-okaidia.css',
+            'sass/styles.scss'
+        ]
+    )
+        .pipe(sass())
         .pipe(rev())
+        .pipe(concat('styles.css'))
         .pipe(gulp.dest('../theme/css/'))
-        .pipe(rev.manifest('../assets.yml', {merge: true}))
+        .pipe(rev.manifest('../assets.yml', {merge : true}))
         .pipe(gulp.dest('./'));
 });
 
-gulp.task('default', ['images', 'scripts', 'styles']);
+// Bundle
+gulp.task('default', ['images', 'icons', 'scripts', 'styles']);
+
+// Watch
+gulp.task('watch', function () {
+    gulp.watch(['sass/**/*.scss'], ['styles']);
```

Comment:
should we have also watcher for js files?

Reply:
Is only relevant for development and additionally I use the "[Gulp Tool Window](https://www.jetbrains.com/help/phpstorm/gulp-tool-window.html)" in PHPStorm.